crates/katana/primitives/src/genesis/json.rs

```diff
@@ -0,0 +1,543 @@
+//! JSON representation of the genesis configuration. Used to deserialize the genesis configuration
+//! from a JSON file.
+
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::{fs, io};
+
+use cairo_lang_starknet::casm_contract_class::{CasmContractClass, StarknetSierraCompilationError};
+use cairo_lang_starknet::contract_class::ContractClass;
+use cairo_vm::types::errors::program_errors::ProgramError;
+use ethers::types::U256;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use starknet::core::types::contract::legacy::LegacyContractClass;
+use starknet::core::types::contract::{ComputeClassHashError, JsonError};
+use starknet::core::types::FromByteArrayError;
+
+use super::{
+    FeeTokenConfig, Genesis, UniversalDeployerConfig, DEFAULT_LEGACY_ERC20_CONTRACT_CASM,
+    DEFAULT_LEGACY_ERC20_CONTRACT_CLASS_HASH, DEFAULT_LEGACY_ERC20_CONTRACT_COMPILED_CLASS_HASH,
+    DEFAULT_LEGACY_UDC_CASM, DEFAULT_LEGACY_UDC_CLASS_HASH, DEFAULT_LEGACY_UDC_COMPILED_CLASS_HASH,
+    DEFAULT_OZ_ACCOUNT_CONTRACT, DEFAULT_OZ_ACCOUNT_CONTRACT_CASM,
+    DEFAULT_OZ_ACCOUNT_CONTRACT_CLASS_HASH, DEFAULT_OZ_ACCOUNT_CONTRACT_COMPILED_CLASS_HASH,
+    DEFAULT_UDC_ADDRESS,
+};
+use crate::block::{BlockHash, BlockNumber, GasPrices};
+use crate::contract::{
+    ClassHash, CompiledContractClass, CompiledContractClassV1, ContractAddress, StorageKey,
+    StorageValue,
+};
+use crate::genesis::{public_key_from_private_key, GenesisAccount, GenesisClass};
+use crate::utils::class::{parse_compiled_class_v0, parse_sierra_class};
+use crate::FieldElement;
+
+#[derive(Debug, PartialEq, Eq, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GenesisClassJson {
+    pub path: PathBuf,
```

Comment:
Can it be any of Sierra or Casm artifact?

Reply:
rn only sierra or casm (legacy).. not sure if we should allow casm (non-legacy) bcs then we wouldn't be able to extract the sierra class from it.